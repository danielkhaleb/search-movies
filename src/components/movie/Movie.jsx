import React, { Component } from 'react';

class Movie extends Component {

    listDetails(source) {
        return (
            <div className="column">
                <div className="card card-movie">
                    <div className="card-image">
                        <figure className="image is-4by3">
                            {
                                source.Poster === undefined || source.Poster === "N/A" ?
                                    // eslint-disable-next-line
                                    <img src={require('../../utils/imgs/image-not-found.gif')} alt="Image not found" />
                                    :
                                    // eslint-disable-next-line 
                                    <img src={source.Poster} alt="Image of movie" />
                            }
                        </figure>
                    </div>
                    <div className="card-content">
                        <div className="media-content">
                            <p className="title is-6">Title: {source.Title}</p>

                        </div>
                        <div class="content">
                            <br />
                            <p className="subtitle is-7">Type: {source.Type}</p>
                            <p className="subtitle is-7">Year: {source.Year}</p>
                            <p className="subtitle is-7">Imdb Id: {source.imdbID}</p>
                            <p className="subtitle is-7">Rated: {source.Rated}</p>
                            <p className="subtitle is-7">Released: {source.Released}</p>
                            <p className="subtitle is-7">Runtime: {source.Runtime}</p>
                            <p className="subtitle is-7">Genre: {source.Genre}</p>
                            <p className="subtitle is-7">Director: {source.Director}</p>
                            <p className="subtitle is-7">Writer: {source.Writer}</p>
                            <p className="subtitle is-7">Actors: {source.Actors}</p>
                            <p className="subtitle is-7">Plot: {source.Plot}</p>
                            <p className="subtitle is-7">Language: {source.Language}</p>
                            <p className="subtitle is-7">Country: {source.Country}</p>
                            <p className="subtitle is-7">Awards: {source.Awards}</p>
                            <p className="subtitle is-7">Metascore: {source.Metascore}</p>
                            <p className="subtitle is-7">imdb Votes: {source.imdbVotes}</p>
                            <p className="subtitle is-7">BoxOffice: {source.BoxOffice}</p>
                            <p className="subtitle is-7">Production: {source.Production}</p>
                            <p className="subtitle is-7">Website: {source.Website}</p>
                        </div>
                    </div>
                </div>
            </div>

        );
    }

    render() {
        const movie = this.listDetails(this.props.data);
        return (
                <div className="columns">
                    {movie}
                </div>
        );
    }
}
export default Movie;
