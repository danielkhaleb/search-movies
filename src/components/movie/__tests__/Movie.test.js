import React from 'react';
import Movie from '../Movie';
import movieJson from "../../../../mock/movie.json"
import renderer from 'react-test-renderer';

describe('Movie component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <Movie
                data={movieJson}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});