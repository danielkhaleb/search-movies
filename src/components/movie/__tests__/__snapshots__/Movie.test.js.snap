// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component renders correctly 1`] = `
<div
  className="columns"
>
  <div
    className="column"
  >
    <div
      className="card card-movie"
    >
      <div
        className="card-image"
      >
        <figure
          className="image is-4by3"
        >
          <img
            alt="Image of movie"
            src="https://m.media-amazon.com/images/M/MV5BMTk3OTM5Njg5M15BMl5BanBnXkFtZTYwMzA0ODI3._V1_SX300.jpg"
          />
        </figure>
      </div>
      <div
        className="card-content"
      >
        <div
          className="media-content"
        >
          <p
            className="title is-6"
          >
            Title: 
            The Notebook
          </p>
        </div>
        <div
          class="content"
        >
          <br />
          <p
            className="subtitle is-7"
          >
            Type: 
            movie
          </p>
          <p
            className="subtitle is-7"
          >
            Year: 
            2004
          </p>
          <p
            className="subtitle is-7"
          >
            Imdb Id: 
            tt0332280
          </p>
          <p
            className="subtitle is-7"
          >
            Rated: 
            PG-13
          </p>
          <p
            className="subtitle is-7"
          >
            Released: 
            25 Jun 2004
          </p>
          <p
            className="subtitle is-7"
          >
            Runtime: 
            123 min
          </p>
          <p
            className="subtitle is-7"
          >
            Genre: 
            Drama, Romance
          </p>
          <p
            className="subtitle is-7"
          >
            Director: 
            Nick Cassavetes
          </p>
          <p
            className="subtitle is-7"
          >
            Writer: 
            Jeremy Leven (screenplay), Jan Sardi (adaptation), Nicholas Sparks (novel)
          </p>
          <p
            className="subtitle is-7"
          >
            Actors: 
            Tim Ivey, Gena Rowlands, Starletta DuPois, James Garner
          </p>
          <p
            className="subtitle is-7"
          >
            Plot: 
            A poor yet passionate young man falls in love with a rich young woman, giving her a sense of freedom, but they are soon separated because of their social differences.
          </p>
          <p
            className="subtitle is-7"
          >
            Language: 
            English
          </p>
          <p
            className="subtitle is-7"
          >
            Country: 
            USA
          </p>
          <p
            className="subtitle is-7"
          >
            Awards: 
            12 wins & 10 nominations.
          </p>
          <p
            className="subtitle is-7"
          >
            Metascore: 
            53
          </p>
          <p
            className="subtitle is-7"
          >
            imdb Votes: 
            468,763
          </p>
          <p
            className="subtitle is-7"
          >
            BoxOffice: 
            $81,000,000
          </p>
          <p
            className="subtitle is-7"
          >
            Production: 
            New Line Cinema
          </p>
          <p
            className="subtitle is-7"
          >
            Website: 
            http://www.thenotebookmovie.com/
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
